subsys/bluetooth/host/smp.c

```diff
@@ -5663,10 +5667,25 @@ int bt_smp_le_oob_generate_sc_data(struct bt_le_oob_sc_data *le_sc_oob)
 	}
 
 	if (!sc_public_key) {
-		err = k_sem_take(&sc_local_pkey_ready, K_FOREVER);
+		err = k_sem_take(&sc_local_pkey_ready, K_NO_WAIT);
+		if (!err) {
+			/* Public key request has finished, but generation failed. Retrying. */
+			err = bt_pub_key_gen(&sc_local_pub_key_cb);
+			if (err && err != -EALREADY) {
+				return err;
+			}
+		}
+
+		/* Public key has been requested but not ready yet. */
+		err = k_sem_take(&sc_local_pkey_ready, K_SECONDS(30));
```

Comment:
We could consider having this timeout as a Kconfig

Reply:
I'd suggest to do this when it becomes a problem. The timeout depends on the platform from what I understand. It is unlikely that we will have to wait here unless if `bt_le_oob_get_local` is called at the start of bluetooth when the public key is not ready yet. I'd rather increase the timeout even more if you want.

Reply:
> I'd suggest to do this when it becomes a problem.

OK with me :)